vme/src/common.h

```diff
@@ -39,11 +39,41 @@ int required_xp(int level);
 int level_xp(int level);
 void set_hits(unit_data *obj, int craftsmanship);
 int skill_point_gain();
+int buy_points(int points, int level, int *error);
 int ability_point_gain(unit_data *ch);
 int ability_point_total(unit_data *ch);
-int distribute_points(sbit16 *skills, int max, int points, int level);
+/**
+ * Algorithm PS 5
+ * Used to distribute points after a given percentage scheme. Each
+ * ability (8 of them) contains the percentage number. The percentage
+ * must add up to 100%.
+ *
+ * @tparam T Type in array
+ * @tparam MAX_ARRAY Max size of array
+ * @param skills Reference to the array to fill
+ * @param max Max size of array
+ * @param points
+ * @param level
+ * @return Returns true if error
+ */
+template<class T, size_t MAX_ARRAY>
+int distribute_points(std::array<T, MAX_ARRAY> &skills, int max, int points, int level)
```

Comment:
Template functions are created when they are used, so that the types match exactly the type being used. Because each size of std::array is a different type, I templated it.

Another solution is to move the functions to a *.tcc file which is included at the end of the normal header

Reply:
Oh, I didn't notice it became a template. Guess there was a good reason the change it from the simple ubit8/16 arrays :-D